httpx_ws/_api.py

```diff
@@ -518,32 +532,20 @@ def _background_keepalive_ping(
     def _wait_until_closed(
         self, callable: typing.Callable[..., TaskResult], *args, **kwargs
     ) -> TaskResult:
-        exit_await = threading.Event()
-
-        def wait_close() -> None:
-            while not exit_await.is_set() and not self._should_close.is_set():
-                time.sleep(0.05)
-
-        executor = concurrent.futures.ThreadPoolExecutor(max_workers=2)
         try:
-            wait_close_task = executor.submit(wait_close)
+            executor, should_close_task = self._get_executor_should_close_task()
             todo_task = executor.submit(callable, *args, **kwargs)
         except RuntimeError as e:
             raise ShouldClose() from e
         else:
             done, _ = concurrent.futures.wait(
-                (todo_task, wait_close_task),  # type: ignore[misc]
+                (todo_task, should_close_task),  # type: ignore[misc]
                 return_when=concurrent.futures.FIRST_COMPLETED,
             )
-            if wait_close_task in done:
+            if should_close_task in done:
                 raise ShouldClose()
             assert todo_task in done
-            if not wait_close_task.cancel():
-                exit_await.set()
-                wait_close_task.result()
             result = todo_task.result()
-        finally:
-            executor.shutdown(False)
```

Comment:
Shouldn't we shutdown `self._executor` somewhere? In `__exit__` method for example?

Reply:
Yes, it's done in [close()](https://github.com/frankie567/httpx-ws/blob/6b89ecefa79205d30e9525635ff8d1b62aefc98c/httpx_ws/_api.py#L442-L443), which is called by [\_\_exit\_\_()](https://github.com/frankie567/httpx-ws/blob/6b89ecefa79205d30e9525635ff8d1b62aefc98c/httpx_ws/_api.py#L124).

Reply:
Oh right! Missed it 😄